src/test/codewhisperer/service/codewhisperer.test.ts

```diff
@@ -66,6 +66,18 @@ describe('codewhisperer', async function () {
         await sendTelemetryEventOptoutCheckHelper(payload, true, false)
     })
 
+    it('sendTelemetryEvent for codeScan for json should respect telemetry optout status', async function () {
+        const payload = {
+            codeScanEvent: {
+                programmingLanguage: { languageName: 'json' },
+                codeScanJobId: anyString(),
+                timestamp: new Date(),
+            },
+        }
+        await sendTelemetryEventOptoutCheckHelper(payload, true, true)
+        await sendTelemetryEventOptoutCheckHelper(payload, true, false)
+    })
```

Comment:
do we seriously need 10 lines of test code for each new filetype?

Reply:
> not only for getDependencyGraph, but also the SendTelemetryEvent with codeScanEvent as the payload.

@andrewyuq requested to add test case to send json as payload to sendTelemetryEvent.
If thats ok I can drop this test case ?

Reply:
The test I am referring to is not related to "telemetry optout status". We don't need to test "telemetry optout status" for different languages. They are independent fields
I was asking for a test that SendTelemetryEvent with codeScanEvent as the payload for the json language, that the request can succeed without any validation error.

Reply:
> I was asking for a test that SendTelemetryEvent with codeScanEvent as the payload for the json language, that the request can succeed without any validation error.

If I understand correctly there is only 1 telemetry event for security scans(`codewhisperer_securityScan`) Previously we tested this event by passing python params. So, correct me if you something like [this](https://github.com/aws/aws-toolkit-vscode/blob/master/src/test/codewhisperer/commands/startSecurityScan.test.ts) to test for json too?

```
        assertTelemetry('codewhisperer_securityScan', {
            codewhispererLanguage: 'json',
            codewhispererCodeScanTotalIssues: 1,
            codewhispererCodeScanIssuesWithFixes: 0,
            codewhispererCodeScanLines: 188,
        } as CodewhispererSecurityScan)
```

Reply:
A manual test is actually fine.

Reply:
Tested manually and this works fine.